src/tls/curve.h

```diff
@@ -2,76 +2,69 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
+#include "crypto/hash.h"
 #include "ds/logger.h"
 #include "tls.h"
 
-#include <secp256k1/include/secp256k1.h>
-#include <secp256k1/include/secp256k1_recovery.h>
+#include <mbedtls/ecp.h>
+#include <openssl/evp.h>
 #include <stdexcept>
 #include <string>
 
+using namespace crypto;
+
 namespace tls
 {
   // SNIPPET_START: supported_curves
-  enum class CurveImpl
+  enum class CurveID
   {
-    secp384r1 = 1,
-    secp256k1_mbedtls = 2,
-    secp256k1_bitcoin = 3,
-
-    service_identity_curve_choice = secp384r1,
+    NONE = 0,
+    SECP384R1,
+    SECP256R1
   };
-  // SNIPPET_END: supported_curves
 
-  // 2 implementations of secp256k1 are available - mbedtls and bitcoin. Either
-  // can be asked for explicitly via the CurveImpl enum. For cases where we
-  // receive a raw 256k1 key/signature/cert only, this flag determines which
-  // implementation is used
-  static constexpr bool prefer_bitcoin_secp256k1 = true;
+  static constexpr CurveID service_identity_curve_choice = CurveID::SECP384R1;
+  // SNIPPET_END: supported_curves
 
   // Helper to access elliptic curve id from context
-  inline mbedtls_ecp_group_id get_ec_from_context(const mbedtls_pk_context& ctx)
+  inline mbedtls_ecp_group_id get_mbedtls_ec_from_context(
+    const mbedtls_pk_context& ctx)
   {
     return mbedtls_pk_ec(ctx)->grp.id;
   }
 
-  // Get mbedtls elliptic curve for given CCF curve implementation
-  inline mbedtls_ecp_group_id get_ec_for_curve_impl(CurveImpl curve)
+  // Get message digest algorithm to use for given elliptic curve
+  inline MDType get_md_for_ec(CurveID ec, bool allow_none = false)
   {
-    switch (curve)
+    switch (ec)
     {
-      case CurveImpl::secp384r1:
-      {
-        return MBEDTLS_ECP_DP_SECP384R1;
-      }
-      case CurveImpl::secp256k1_mbedtls:
-      case CurveImpl::secp256k1_bitcoin:
-      {
-        return MBEDTLS_ECP_DP_SECP256K1;
-      }
+      case CurveID::SECP384R1:
+        return MDType::SHA384;
+      case CurveID::SECP256R1:
+        return MDType::SHA256;
       default:
       {
-        throw std::logic_error(
-          "Unhandled curve type: " +
-          std::to_string(static_cast<size_t>(curve)));
+        if (allow_none)
```

Comment:
When do we use allow_none?

Reply:
In `verify_hash`, but I don't like it either, I'll see if I can get rid of it.